### PR TITLE
Improve assign flags form redirects

### PR DESCRIPTION
**Pull request by eadpearce**

### CODE REVIEW

File: caseworker/flags/views.py [hunk 1]
```diff
@@ -288,7 +287,7 @@ def get_level(self):
             return FlagLevel.CASES
         elif self.request.GET.get("organisation"):
             return FlagLevel.ORGANISATIONS
-        elif self.request.GET.get("goods") or self.request.GET.get("goods_types"):
+        elif self.request.GET.get("good_ids") or self.request.GET.get("goods_types"):
```
Comment (richtier) [inline caseworker/flags/views.py:290]: do we want this change? I thought we reverted back to `goods`

File: unit_tests/conftest.py [hunk 2]
```diff
@@ -516,25 +516,27 @@ def data_standard_case():
...
-                        "sub_type_other": None,
-                        "role": {"key": "other", "value": "Other"},
-                        "role_other": None,
-                        "flags": [],
-                        "copy_of": None,
-                        "deleted_at": None,
-                        "clearance_level": None,
-                        "descriptors": None,
-                    },
+                    "data": [
```
Comment (richtier) [inline unit_tests/conftest.py:519]: hmm, how did tests pass before this changed data type I wonder

File: unit_tests/caseworker/flags/test_views.py [hunk 3]
```diff
@@ -0,0 +1,131 @@
+import pytest
```
Comment (richtier) [inline unit_tests/caseworker/flags/test_views.py:1]: dope tests there

File: caseworker/flags/views.py (hunk 1 above)
Comment (eadpearce) [reply caseworker/flags/views.py:290]: ah good spot. i will change this back

File: unit_tests/conftest.py (hunk 2 above)
Comment (eadpearce) [reply unit_tests/conftest.py:519]: my guess is this part of the data just wasn't accessed by tests before now

**Review by richtier — APPROVED** [review]

**Review by r4vi — APPROVED** [review]

**Pull request merged**